src/transformers/models/distilbert/modeling_distilbert.py

```diff
@@ -92,7 +93,11 @@ def __init__(self, config):
         self.LayerNorm = nn.LayerNorm(config.dim, eps=1e-12)
         self.dropout = nn.Dropout(config.dropout)
 
-    def forward(self, input_ids):
+        # position_ids (1, len position emb) is contiguous in memory and exported when serialized
+        self.register_buffer("position_ids", torch.arange(config.max_position_embeddings).expand((1, -1)))
+        self.position_embedding_type = getattr(config, "position_embedding_type", "absolute")
```

Comment:
If we add this then when loading distilbert model it will print warning  like this
`Some weights of DistilBertModel were not initialized from the model checkpoint at distilbert-base-uncased and are newly initialized: ['distilbert.embeddings.position_ids']`.

Also, distilbert can use sinusoidal positional embedding which have no hard limit on maximum sequence length.

So I would remove this and keep the logic as it is for this model.

Reply:
Oh I see, but in that case how do we initialise self.position_ids which are used by line 112 below?

Reply:
distillbert computes the `position_ids` https://github.com/huggingface/transformers/blob/e783ea730456f67a246cb6d5db4893349f3e5af2/src/transformers/models/distilbert/modeling_distilbert.py#L104